gammapy/background/energy_offset_array.py

```diff
@@ -19,8 +42,9 @@ class EnergyOffsetArray(object):
          energy bin vector
     offset : `~astropy.coordinates.Angle`
         offset bin vector
-    data : `~numpy.ndarray`
-        data array (2D)
+    data : `~astropy.units.Quantity`
+        data array (2D): Background rate
+        units: s-1 TeV-1 sr-1
```

Comment:
I clean the write and read methods and ass a test!
For the moment I have a problem with the Table from Astropy in the method to_table… the column of the table must have the same length and this is not the case for the offset or the energy. 
Le 9 févr. 2016 à 21:01, Lea Jouvin <ljouvin@apc.in2p3.fr> a écrit :

> Let’s do this(-:
> Le 9 févr. 2016 à 20:36, Christoph Deil <notifications@github.com> a écrit :
> 
>> In gammapy/background/energy_offset_array.py:
>> 
>> > @@ -19,8 +42,9 @@ class EnergyOffsetArray(object):
>> >           energy bin vector
>> >      offset : `~astropy.coordinates.Angle`
>> >          offset bin vector
>> > -    data : `~numpy.ndarray`
>> > -        data array (2D)
>> > +    data : `~astropy.units.Quantity`
>> > +        data array (2D): Background rate
>> > +        units: s-1 TeV-1 sr-1
>> This class should be able to represent counts, livetime, rate (maybe other things like PSF containment in the future).
>> 
>> So in this class, you shouldn't put rate in s-1 TeV-1 sr-1 as default unit.
>> 
>> Either just use plain numpy array, or put a dimensionless quantity:
>> http://astropy.readthedocs.org/en/latest/units/standard_units.html#the-dimensionless-unit
>> 
>> I don't have experience with dimensionless quantity and FITS I/O. Probably it will take some trial and error to make this work nicely. Maybe just put whatever you like for now, and then we'll continue discussing units in the other pull request, where you have counts, time and rate cubes and we see what actually works well...?
>> 
>> —
>> Reply to this email directly or view it on GitHub.
>> 
>